tests/regression/diffkabi/diffkabi_test.py

```diff
@@ -54,61 +58,53 @@ def __init__(self, spec):
-                                           debug=spec.debug)
-        second_builder.source.build_cscope_database()
-
-        # Build the modules
-        mod_first = first_builder.build_llvm_function(function)
-        mod_second = second_builder.build_llvm_function(function)
-
-        # Add the function to the dictionary
-        fun = (function, mod_first, mod_second)
-        spec.built_functions[fun] = Result[desc.upper()]
-
-        # The modules were already built when finding their sources.
-        # Now the files need only to be copied to the right place.
-        #
-        # Note: the files are copied to the task directory for reference only.
-        # The exact name of the module is not known before building the
-        # function, therefore building from module sources in kernel_modules or
-        # using already built LLVM IR files is not possible.
-        mod_path_old = mod_first.llvm
-        mod_path_new = mod_second.llvm
-        mod_old = os.path.basename(mod_first.name)
-        mod_new = os.path.basename(mod_second.name)
-        task_dir = os.path.join(tasks_path, mod_old)
-        if not os.path.isdir(tasks_path):
-            os.mkdir(tasks_path)
-        if not os.path.isdir(task_dir):
-            os.mkdir(task_dir)
-        old_src = os.path.join(task_dir, mod_old + "_old.c")
-        new_src = os.path.join(task_dir, mod_new + "_new.c")
-        old_llvm = os.path.join(task_dir, mod_old + "_old.ll")
-        new_llvm = os.path.join(task_dir, mod_new + "_new.ll")
-        old_source_src = os.path.splitext(mod_path_old)[0] + ".c"
-        new_source_src = os.path.splitext(mod_path_new)[0] + ".c"
-
-        shutil.copyfile(mod_path_old, old_llvm)
-        shutil.copyfile(old_source_src, old_src)
-
-        shutil.copyfile(mod_path_new, new_llvm)
-        shutil.copyfile(new_source_src, new_src)
+    # Find the modules
+    first_builder = LlvmKernelBuilder(spec.old_kernel, None,
+                                      debug=spec.debug)
+    second_builder = LlvmKernelBuilder(spec.new_kernel, None,
+                                       debug=spec.debug)
+
+    # Build the modules
+    old_module = first_builder.build_file_for_function(spec.function)
+    new_module = second_builder.build_file_for_function(spec.function)
+
+    # The modules were already built when finding their sources.
+    # Now the files need only to be copied to the right place.
+    #
+    # Note: the files are copied to the task directory for reference only.
+    # The exact name of the module is not known before building the
+    # function, therefore building from module sources in kernel_modules or
+    # using already built LLVM IR files is not possible.
+    mod_old = os.path.basename(old_module.name)
+    mod_new = os.path.basename(new_module.name)
+    spec.task_dir = os.path.join(tasks_path, spec.function)
+    if not os.path.isdir(tasks_path):
+        os.mkdir(tasks_path)
```

Comment:
Is this necessary? We could suppose that this directory exists.

Reply:
I believe it is, because the directory isn't supposed to contain any files, therefore it isn't in the Git repository, so it has to be created.

Reply:
I see, then let's do this somewhere else, so that we don't have to check its existence in each test spec.